core/constants.py

```diff
@@ -59,6 +60,30 @@ def remove_comments(text: str) -> str:
     return re.sub(r'  //.*\n', r'', text)
 
 
+def get_package_file_contents(package: str, filepath: str) -> str:
+    """Open file and return its contents. This needs to be used for files that
+    are loaded by the Python code directly, like constants.ts or
+    rich_text_components.json. This function is needed to make loading these
+    files work even when Oppia is packaged.
+
+    Args:
+        package: str. The package where the file is located.
+            For Oppia the package is usually the folder in the root folder,
+            like 'core' or 'extensions'.
+        filepath: str. The path to the file in the package.
+
+    Returns:
+        str. The contents of the file.
+    """
+    try:
+        file = io.open(os.path.join(package, filepath), 'r', encoding='utf-8')
+        return file.read()
+    except FileNotFoundError:
+        data = pkgutil.get_data(package, filepath)
+        assert isinstance(data, bytes)
+        return data.decode('utf-8')
```

Comment:
I did try that but it does not work, this error pops up: `module core.utils has no attribute get_package_file_contents`
I think there is some sort of race condition going on here because the file `constants` is called very early which calls `utils` but `utils` is not loaded at that point yet

Reply:
Ah I see, can you please check what values for constants we use in utils?

Reply:
@vojtechjelinek utils uses a lot of constants, a quick search gives me around 18 results. eg constants.INVALID_NAME_CHARS, VALID_URL_FRAGMENT_REGEX... etc

Reply:
Hmm, okay. Lets just keep this function here, but lets explain why is it here and why it cannot be anywhere else.

Reply:
Added a comment PTAL